tests/unit/amo/components/TestConfirmationDialog.js

```diff
@@ -1,118 +1,85 @@
 import * as React from 'react';
+import userEvent from '@testing-library/user-event';
 
-import Button from 'amo/components/Button';
-import ConfirmationDialog, {
-  ConfirmationDialogBase,
-} from 'amo/components/ConfirmationDialog';
-import {
-  createFakeEvent,
-  fakeI18n,
-  shallowUntilTarget,
-} from 'tests/unit/helpers';
+import ConfirmationDialog from 'amo/components/ConfirmationDialog';
+import { render as defaultRender, screen } from 'tests/unit/helpers';
 
 describe(__filename, () => {
-  function render(otherProps = {}) {
-    const props = {
-      i18n: fakeI18n(),
-      onCancel: sinon.stub(),
-      onConfirm: sinon.stub(),
-      ...otherProps,
-    };
-    return shallowUntilTarget(
-      <ConfirmationDialog {...props} />,
-      ConfirmationDialogBase,
-    );
+  function render(props = {}) {
+    return defaultRender(<ConfirmationDialog {...props} />);
   }
 
   it('renders a dialog', () => {
     const message = 'this action is risky, are you sure?';
-    const root = render({ message });
+    render({ message });
 
-    expect(root.find('.ConfirmationDialog-message')).toHaveLength(1);
-    expect(root.find('.ConfirmationDialog-message')).toHaveText(message);
+    expect(screen.getByText(message)).toHaveClass('ConfirmationDialog-message');
 
-    expect(root.find(Button)).toHaveLength(2);
+    const confirmButton = screen.getByRole('button', { name: 'Confirm' });
+    expect(confirmButton).toHaveClass('ConfirmationDialog-confirm-button');
```

Comment:
In this case we can argue for this assertion because, although the class name is hardcoded, it is passed as a property from `ConfirmationDialog` to `Button`, so asserting that we see the expected class makes sense as it is part of the logic of the `Button`.